src/libs/workspace-utils.ts

```diff
@@ -29,6 +29,11 @@ interface BaseWorkspaceInfo {
   createdDate: string;
   createdBy: string;
   lastModified: string;
+  bucketName?: string;
+  attributes?: Record<string, any>;
```

Comment:
Will `unknown` work here? That way, we get the reminder that we need to check what type the attribute is before using it. `any` will let us treat it as any type.
https://www.typescriptlang.org/docs/handbook/2/functions.html#unknown

```suggestion
  attributes?: Record<string, unknown>;
```

Reply:
Also, I'm hesitant about declaring these fields optional. I understand different subsets of fields may be retrieved from the API, but I think we need to figure out a way to deal with that for _all_ workspace fields instead of only some.

I think it will be easier to deal with that if the base type declares fields as required. Then components that need a subset of Workspace can slice that type down to what they need.